python/pyspark/sql/dataframe.py

```diff
@@ -1750,6 +1751,29 @@ def _to_scala_map(sc, jm):
     return sc._jvm.PythonUtils.toScalaMap(jm)
 
 
+def _to_numpy_type(dt):
+    """
+    Convert a Spark SQL data type to Pandas data type.
+    """
+    import numpy as np
+    if type(dt) == BooleanType:
+        return 'bool'
+    elif type(dt) == ByteType:
+        return 'int8'
+    elif type(dt) == ShortType:
+        return 'int16'
+    elif type(dt) == IntegerType:
+        return 'int32'
+    elif type(dt) == LongType:
+        return 'int64'
+    elif type(dt) == FloatType:
+        return 'float32'
+    elif type(dt) == DoubleType:
+        return 'float64'
+    else:
+        return 'object'
```

Comment:
I think it might cause problems to have all non-primitive types as object.  Things like timestamps will be inferred form a datetime object, for example: 

```python
In [10]: pdf = pd.DataFrame.from_records([(1.0, 1, "a", datetime.datetime.now())])

In [11]: pdf.dtypes
Out[11]: 
0           float64
1             int64
2            object
3    datetime64[ns]
dtype: object

In [12]: pdf.astype({0: "float32", 1: "int32", 2: "object", 3: "object"}).dtypes
Out[12]: 
0    float32
1      int32
2     object
3     object
dtype: object
```